From: Benjamin Drung <benjamin.drung@ionos.com>
Date: Wed, 17 Nov 2021 15:28:50 +0100
Subject: docs: Support building documentation locally

Building the documentation locally without Internet access will fail,
because sphinx will try to download the intersphinx inventory from
http://docs.python.org/3/objects.inv.

Support building the documentation locally without Internet access by
searching for locally installed intersphinx inventory first.

Bug-Debian: https://bugs.debian.org/755026
Forwarded: https://github.com/saltstack/salt/pull/61248
Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>
---
 doc/conf.py | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/doc/conf.py b/doc/conf.py
index e8e1612..4b96bae 100644
--- a/doc/conf.py
+++ b/doc/conf.py
@@ -33,7 +33,17 @@ import salt.version  # isort:skip
 formulas_dir = os.path.join(os.pardir, docs_basepath, "formulas")
 
 # ----- Intersphinx Settings ------------------------------------------------>
-intersphinx_mapping = {"python": ("https://docs.python.org/3", None)}
+intersphinx_mapping = {
+    "python": (
+        "https://docs.python.org/3",
+        (
+            "/usr/share/doc/python{}.{}/html/objects.inv".format(
+                sys.version_info[0], sys.version_info[1]
+            ),
+            None,
+        ),
+    )
+}
 # <---- Intersphinx Settings -------------------------------------------------
 
 # -- General Configuration -----------------------------------------------------
